src/test/java/io/neonbee/test/helper/MultipartResponse.java

```diff
@@ -0,0 +1,182 @@
+package io.neonbee.test.helper;
+
+import static io.vertx.core.MultiMap.caseInsensitiveMultiMap;
+import static io.vertx.core.http.HttpHeaders.CONTENT_TYPE;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Iterables;
+
+import io.vertx.core.MultiMap;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.ext.web.client.HttpResponse;
+
+/**
+ * Represents boundary and parts of an <a href=
+ * "http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_MultipartBatchResponse">OData
+ * Multipart Batch Response</a>.
+ */
+public class MultipartResponse {
+    private final String boundary;
+
+    private final List<Part> parts;
+
+    public MultipartResponse(String boundary, List<Part> parts) {
+        this.boundary = boundary;
+        this.parts = parts;
+    }
+
+    public String getBoundary() {
+        return boundary;
+    }
+
+    public List<Part> getParts() {
```

Comment:
Couldn't we use ODataResponse from Olingo instead of introducing Part? Then we can maybe also using the Olingo deserializer.

Reply:
As discussed, there is no deserializer for batch responses.